pkg/controllers/selection/controller.go

```diff
@@ -142,11 +142,11 @@ func validateAffinity(p *v1.Pod) (errs error) {
 	if p.Spec.Affinity == nil {
 		return nil
 	}
-	if pod.HasPodAffinity(p) {
-		errs = multierr.Append(errs, fmt.Errorf("pod affinity is not supported"))
+	if pod.HasRequiredPodAffinity(p) {
+		errs = multierr.Append(errs, fmt.Errorf("pod affinity rule 'requiredDuringSchedulingIgnoreDuringExecution' is not supported"))
```

Comment:
Is this meant to say 'is not supported'?

https://github.com/aws/karpenter/pull/1541/files#diff-a6635dd6ddb6e0a211850401eb1275a378603ffa15afdcb98c2ba41ab624476dR329

Seems to indicate that you are stating it should be supported in the tests. The same applies to pod Anti Affinity.

Reply:
Actually reading the comments above looks like this is intentional...

Reply:
Sorry, I didn't follow this comment. This change was making Karpenter ignore preferred pod affinity, while still rejecting scheduling of required pod affinity entirely.

Reply:
This may be more simply achieved by implementing using preferences.go

Reply:
Sure, happy that give a try!

Reply:
@ellistarn I included pod affinity and anti-affinity preferences relaxation, but I had to maintain the affinity validation as it's proposed in this PR (otherwise the code would return [here](https://github.com/aws/karpenter/blob/14d84eb8ba231fe5966346936374b2ed20b545ca/pkg/controllers/selection/controller.go#L74), and we won't be able to reach `Preferences.Relax` as you were suggesting [here](https://github.com/aws/karpenter/pull/1541#discussion_r830330696)).

Something that I'm not 100% convinced about, is that currently Karpenter doesn't take into consideration any podAffinity / podAntiAffinity to match with a provisioner, so it sounds a bit irrelevant to iteratively remove pod affinity / anti-affinity preferences, it looks to me that it only adds overhead without bringing much value. 

What your thoughts are on this one?

Reply:
There shouldn't be too many terms, and it will hopefully align well with future support for required pod affinity/anti-affinity.

Reply:
thank you @tzneal for your quick reply! :)